cost-model/src/transaction_cost.rs

```diff
@@ -1,10 +1,109 @@
-use solana_sdk::pubkey::Pubkey;
+use {crate::block_cost_limits, solana_sdk::pubkey::Pubkey};
+
+/// TransactionCost is used to represent resources required to process
+/// a transaction, denominated in CU (eg. Compute Units).
+/// Resources required to process a regular transaction often include
+/// an array of variables, such as execution cost, loaded bytes, write
+/// lock and read lock etc.
+/// Vote has a simpler and pre-determined format, it's cost structure
+/// can be simpler, calculation quicker.
+#[derive(Debug)]
+pub enum TransactionCost {
+    Vote { writable_accounts: Vec<Pubkey> },
+    Transaction(UsageCostDetails),
+}
+
+impl TransactionCost {
+    pub fn sum(&self) -> u64 {
+        match self {
+            Self::Vote { writable_accounts } => {
+                let num_of_signatures = writable_accounts.len() as u64;
+
+                solana_vote_program::vote_processor::DEFAULT_COMPUTE_UNITS
+                    .saturating_add(
+                        block_cost_limits::SIGNATURE_COST.saturating_mul(num_of_signatures),
+                    )
+                    .saturating_add(
+                        block_cost_limits::WRITE_LOCK_UNITS.saturating_mul(num_of_signatures),
+                    )
+            }
+            Self::Transaction(usage_cost) => usage_cost.sum(),
+        }
+    }
+
+    pub fn bpf_execution_cost(&self) -> u64 {
+        match self {
+            Self::Vote { .. } => 0,
+            Self::Transaction(usage_cost) => usage_cost.bpf_execution_cost,
+        }
+    }
+
+    pub fn is_simple_vote(&self) -> bool {
+        match self {
+            Self::Vote { .. } => true,
+            Self::Transaction(_) => false,
+        }
+    }
+
+    pub fn data_bytes_cost(&self) -> u64 {
+        match self {
+            Self::Vote { .. } => 0,
+            Self::Transaction(usage_cost) => usage_cost.data_bytes_cost,
+        }
+    }
+
+    pub fn account_data_size(&self) -> u64 {
+        match self {
+            Self::Vote { .. } => 0,
```

Comment:
probably we want comments here. The actual size is not 0 for loaded accounts. It's I think we just don't want t include the costs for votes

Reply:
once decided to give simple vote a static cost, maybe of this can be refactored away.